### PR TITLE
feat: A/B buttons input events

### PR DESCRIPTION
**Pull request by oOZakMcKrackeNOo**
Retrieve and report A/B buttons input events to kernel.

From client side, we need:

- Create an InputManager that retrieves the events from any input device.
- Create an event to send that information to explorer.
- Report the event to explorer.

From kernel side, we need:

- Create a new event type for A/B buttons.
- Allow the users to subscribe/unsubscribe to those events to input object.
- Get the event from Unity and dispatch it to the input object.

#417

### CODE REVIEW

**Review by eordano — APPROVED** [review]

File: packages/decentraland-ecs/src/decentraland/Input.ts [hunk 1]
```diff
@@ -28,7 +29,7 @@ export type InputState = Record<
 export type LocalPointerEvent = GlobalInputEventResult & {
```
Comment (moliva) [inline packages/decentraland-ecs/src/decentraland/Input.ts]: With the action button refactor, this name becomes inconsistent. Suggestion, rename to `LocalActionButtonEvent`.
```suggestion
export type LocalActionButtonEvent = GlobalInputEventResult & {
```

File: packages/decentraland-ecs/src/decentraland/Input.ts [hunk 2]
```diff
@@ -28,7 +29,7 @@ export type InputState = Record<
...
   origin: Vector3
   direction: Vector3
-  pointer: Pointer
+  pointer: ActionButton
```
Comment (moliva) [inline packages/decentraland-ecs/src/decentraland/Input.ts]: Same as above:
```suggestion
  button: ActionButton
```

File: packages/decentraland-ecs/src/decentraland/Input.ts [hunk 3]
```diff
@@ -76,16 +87,29 @@ export class Input {
     return this.internalState
   }
 
-  private subscriptions: Record<InputEventKind, Array<(e: LocalPointerEvent) => void>> = {
-    BUTTON_DOWN: [],
-    BUTTON_UP: []
+  private subscriptions: Record<ActionButton, Record<InputEventKind, Array<Subscription>>> = {
```
Comment (moliva) [inline packages/decentraland-ecs/src/decentraland/Input.ts:86]: nit: Initialize with a foreach statement.
```suggestion
  private subscriptions: Record<ActionButton, Record<InputEventKind, Array<Subscription>>> = Object.keys(ActionButton).map(button => ({ 
  [button]: {
      BUTTON_DOWN: [],
      BUTTON_UP: []
    }})).reduce((accumulator, current) => ({ ...accumulator, ...current}))
```

File: packages/decentraland-ecs/src/decentraland/Input.ts [hunk 4]
```diff
@@ -76,16 +87,29 @@ export class Input {
...
+    },
+    [ActionButton.SECONDARY]: {
+      BUTTON_DOWN: [],
+      BUTTON_UP: []
+    }
   }
 
   private internalState: InputState = {
-    [Pointer.PRIMARY]: {
+    [ActionButton.POINTER]: {
```
Comment (moliva) [inline packages/decentraland-ecs/src/decentraland/Input.ts:102]: nit: Same as above:
```
Object.keys(ActionButton).map(button => ({ 
  [button]: {
      BUTTON_DOWN: false
    }})).reduce((accumulator, current) => ({ ...accumulator, ...current}))
```

File: packages/decentraland-ecs/src/decentraland/Input.ts [hunk 5]
```diff
@@ -103,76 +127,122 @@ export class Input {
...
+   * @param pointerId - The id of the button.
+   * @param useRaycast - Enables getting raycast information.
    * @param fn - A callback function to be called when the event is triggered.
    */
-  public subscribe(eventName: InputEventKind, fn: (e: LocalPointerEvent) => void) {
-    this.subscriptions[eventName].push(fn)
-    return () => this.unsubscribe(eventName, fn)
+  public subscribe(
+    eventName: InputEventKind,
+    pointerId: ActionButton,
```
Comment (moliva) [inline packages/decentraland-ecs/src/decentraland/Input.ts]: Change for actionId or button to be consistent:
```suggestion
    button: ActionButton,
```

**Review by moliva — COMMENTED** [review]
Leaving some comments, mostly renames and nitpicks to avoid having to add initialization code with new buttons in the future.

**Review by pravusjif — APPROVED** [review]

**Review by moliva — APPROVED** [review]

**Review by moliva — APPROVED** [review]

**Pull request merged**